mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb

```diff
@@ -746,18 +751,46 @@ func ResourceComposerEnvironment() *schema.Resource {
 									"task_logs_retention_config": {
 										Type:         schema.TypeList,
 										Description: `Optional. The configuration setting for Task Logs.`,
-										Required:     true,
+										Optional:     true,
+										ForceNew:     false,
+										AtLeastOneOf: composerDataRetentionKeys,
 										Elem: &schema.Resource{
 											Schema: map[string]*schema.Schema{
 												"storage_mode": {
 													Type:         schema.TypeString,
 													Optional:     true,
+													ForceNew:     false,
```

Comment:
Seems like the force new is happening due to the resource using the default value from the api perhaps?

```
                 ~ data_retention_config {
                      ~ airflow_metadata_retention_config {
                          - retention_days = 100 -> null
                          ~ retention_mode = "RETENTION_MODE_ENABLED" -> "RETENTION_MODE_DISABLED"
                        }
        
                        # (1 unchanged block hidden)
                    }
```

Reply:
I have more or less the idea what is wrong here. I will notice you asap when I update. 
Sorry for the delay.

Creations are working properly.
Updates have a bug.